src/reflect/scala/reflect/internal/util/Collections.scala

```diff
@@ -158,6 +158,19 @@ trait Collections {
     if (lb eq null) Nil else lb.result
   }
 
+  // compare to foldLeft[A, B](xs)
+  final def foldLeft2[A1, A2, B](xs1: List[A1], xs2: List[A2])(z0: B)(f: (B, A1, A2) => B): B = {
+    var ys1 = xs1
+    var ys2 = xs2
+    var res = z0
+    while (!ys1.isEmpty && !ys2.isEmpty) {
```

Comment:
attn @som-snytt

Reply:
I'm a quick study. Diego didn't have my disadvantages to overcome.